components/script/dom/mutationobserver.rs

```diff
@@ -37,4 +72,159 @@ impl MutationObserver {
         ScriptThread::add_mutation_observer(&*observer);
         Ok(observer)
     }
+
+    /// https://dom.spec.whatwg.org/#queue-a-mutation-observer-compound-microtask
+    /// Queue a Mutation Observer compound Microtask.
+    pub fn queueMutationObserverCompoundMicrotask() {
+        // Step 1
+        if ScriptThread::is_mutation_observer_compound_microtask_queued() {
+            return;
+        }
+        // Step 2
+        ScriptThread::set_mutation_observer_compound_microtask_queued(true);
+        // Step 3
+        let compoundMicrotask = Microtask::NotifyMutationObservers();
+        ScriptThread::enqueue_microtask(compoundMicrotask);
+    }
+
+    /// https://dom.spec.whatwg.org/#notify-mutation-observers
+    /// Notify Mutation Observers.
+    pub fn notifyMutationObservers() -> Fallible<()> {
+        // Step 1
+        ScriptThread::set_mutation_observer_compound_microtask_queued(false);
+        // Step 2
+        let notifyList = ScriptThread::get_mutation_observer();
+        // Step 3, Step 4 not needed as Servo doesn't implement anything related to slots yet.
+        // Step 5: Ignore the specific text about execute a compound microtask.
+        for (i, mo) in notifyList.iter().enumerate() {
+            let queue = mo.record_queue.clone();
+            *mo.record_queue.borrow_mut() = Vec::new();
+            // TODO: Step 5.3 Remove all transient registered observers whose observer is mo.
+            if !queue.borrow().is_empty() {
+                let callback: MutationCallback = *mo.callback.deref();
+                return callback.Call_(mo, *queue.borrow_mut().deref(), mo, ExceptionHandling::Report);
```

Comment:
I have removed the return statement, but I still get the following error on this line:
```
error[E0277]: the trait bound `dom::bindings::js::Root<dom::mutationobserver::MutationObserver>: dom::bindings::reflector::DomObject` is not satisfied
   --> /home/sumit/Dropbox/NCSU/OODD/FinalProject/servo/components/script/dom/mutationobserver.rs:106:26
    |
106 |                 callback.Call_(&*mo, queue, &*mo, ExceptionHandling::Report);
    |                          ^^^^^ the trait `dom::bindings::reflector::DomObject` is not implemented for `dom::bindings::js::Root<dom::mutationobserver::MutationObserver>`

error: aborting due to previous error
```

This is the only error that is shown now.